src/Akka.Persistence.Sql/Query/Dao/BaseByteReadArrayJournalDao.cs

```diff
@@ -239,35 +246,51 @@ private static int MaxTake(long max)
             var maxTake = MaxTake(max);
 
             return AsyncSource<JournalRow>.FromEnumerable(
-                new { _connectionFactory = ConnectionFactory, maxTake, maxOffset, offset },
+                new { _connectionFactory = ConnectionFactory, args=new QueryArgs(offset,maxOffset,maxTake,_tagMode) },
                 async input =>
                 {
-                    return await input._connectionFactory.ExecuteWithTransactionAsync(
-                        ReadIsolationLevel,
-                        ShutdownToken,
-                        async (connection, token) =>
-                        {
-                            var query = connection
-                                .GetTable<JournalRow>()
-                                .Where(
-                                    r =>
-                                        r.Ordering > input.offset &&
-                                        r.Ordering <= input.maxOffset &&
-                                        r.Deleted == false)
-                                .OrderBy(r => r.Ordering)
-                                .Take(input.maxTake);
-
-                            return await AddTagDataIfNeededAsync(query, connection, token);
-                        });
+                    var cf = input._connectionFactory;
+                    var a = input.args;
+                    return await ExecuteEventQuery(a);
                 }
             ).Via(_deserializeFlow);
         }
+        
+        
+        internal async Task<List<JournalRow>> ExecuteEventQuery(QueryArgs queryArgs)
```

Comment:
This is intentionally moved into it's own method,

Among other things, Another memory optimization for us to consider is -not- using Akka streams in the `Unfold` internal read calls; Where this gets tricky, is that marshalling via Akka streams pipeline, neatly solves problems where a lot of DB Providers aren't really all that async 🙃. 

SQLite (Fair enough,) I think some oracle providers, IDK about MySQL...

I can fold back in for now if needed, but it does make the pipeline at least a little bit neater.